### PR TITLE
Fixing .htconfig loading

### PR DESCRIPTION
**Pull request by nupplaphil**
Fixing legacy .htconfig loading, described here
https://friendica.philipp.info/display/735a2029-435c-8d17-ec92-b96564712636

I reverted #6828 because in the legacy `.htconfig.php` file, the entry `$a->config['system']['db_charset']` was used.

### CODE REVIEW

File: src/Core/Config/Cache/ConfigCacheLoader.php [hunk 1]
```diff
@@ -103,47 +103,64 @@ private function loadLegacyConfig($name)
...
 		$filePath = $this->baseDir  . DIRECTORY_SEPARATOR . '.' . $name . '.php';
 
+		$config = [];
+
 		if (file_exists($filePath)) {
 			$a = new \stdClass();
 			$a->config = [];
 			include $filePath;
 
+			$htconfigAr = array_keys($a->config);
```
Comment (MrPetovan) [inline src/Core/Config/Cache/ConfigCacheLoader.php]: Please rename to `$htconfigCategories`.

File: src/Core/Config/Cache/ConfigCacheLoader.php [hunk 2]
```diff
@@ -103,47 +103,64 @@ private function loadLegacyConfig($name)
...
+
+			// map the legacy configuration structure to the current structure
+			foreach ($htconfigAr as $htconfig) {
```
Comment (MrPetovan) [inline src/Core/Config/Cache/ConfigCacheLoader.php]: Please rename `$htconfig` to `$htconfigCategory`.

File: src/Core/Config/Cache/ConfigCacheLoader.php [hunk 3]
```diff
@@ -103,47 +103,64 @@ private function loadLegacyConfig($name)
...
+				if (isset($a->config[$htconfig]) && is_array($a->config[$htconfig])) {
```
Comment (MrPetovan) [inline src/Core/Config/Cache/ConfigCacheLoader.php]: `isset($a->config[$htconfig])` seems to be superfluous as `$htconfig` comes from `array_keys($a->config)` so `$a->config[$htconfig]` will always be set.

File: src/Core/Config/Cache/ConfigCacheLoader.php (hunk 1 above)
Comment (nupplaphil) [reply src/Core/Config/Cache/ConfigCacheLoader.php]: Got it :-P ;-)

**Pull request merged**